### PR TITLE
Camb dark energy model

### PR DESCRIPTION
**Pull request by t-ferreira**
I included an option to choose which type of dark energy you want when using camb. It defaults to "fluid", now one can easily set it to "ppf".

I simply set "self._params.DE_model_camb = DE_model_camb" because I was having a difficult time with the C files.

### CODE REVIEW

**Comment by damonge** [comment]
@tilmantroester would your PR (#854) be indirectly addressing this? We want to avoid having too many different ways of doing the same thing

**Comment by tilmantroester** [comment]
Yes, in the sense that the dark energy model could be specified in `extra_parameters["camb"]`.

**Comment by t-ferreira** [comment]
@tilmantroester I could do that, if you wouldn't mind.

**Comment by t-ferreira** [comment]
@tilmantroester I added a test to check for a w crossing -1, if ppf is not selected as the dark energy model.

File: pyccl/tests/test_power.py [hunk 1]
```diff
@@ -572,3 +572,21 @@ def test_input_nonlin_raises():
...
+        ccl.linear_matter_power(cosmo, 1, 1)
+
+
+def test_camb_correct_de_model():
+    """Check that w is not less than -1, if the chosen dark energy model for
+    CAMB is fluid."""
+    cosmo = ccl.CosmologyVanillaLCDM(
+        transfer_function='boltzmann_camb', w0=-1, wa=-1)
+    with pytest.raises(ValueError):
+        ccl.linear_matter_power(cosmo, 1, 1)
```
Comment (tilmantroester) [inline pyccl/tests/test_power.py:592]: Apologies for the delay getting back to this. Maybe just add a
```python
cosmo = ccl.CosmologyVanillaLCDM(
        transfer_function='boltzmann_camb', w0=-1, wa=-1, extra_parameters={"camb": {"dark_energy_model": "ppf"}}
)
ccl.linear_matter_power(cosmo, 1, 1)
```
to check that the PPF version is running? That would directly check for the case of the issue that prompted this (#779).

**Comment by c-d-leonard** [comment]
@tilmantroester could you have a look? looks like Tassia may have added the most recent check you asked about

**Comment by tilmantroester** [comment]
Looks good to me!

**Review by tilmantroester — APPROVED** [review]

**Pull request merged**